### PR TITLE
feat: Add fill constraint

### PR DESCRIPTION
**Pull request by kdheepak**

### CODE REVIEW

File: src/layout.rs [hunk 1]
```diff
@@ -31,6 +32,9 @@ macro_rules! constraint {
     ( == $expr:expr ) => {
         ratatui::prelude::Constraint::Length($expr)
     };
+    ( ~ $expr:expr ) => {
+        ratatui::prelude::Constraint::Fill($expr)
```
Comment (EdJoPaTo) [inline src/layout.rs:36]: Using the actual path here is more clear what is happening. But I'm already known for picking on prelude I guess :ghost: 

```suggestion
        ratatui::layout::Constraint::Fill($expr)
```

Comment (kdheepak) [reply src/layout.rs]: Done!

**Comment by joshka** [comment]
I'm not a fan at all of this particular macro as it feels like "too much magic" to me, so I have no opinion on merging this.

File: src/layout.rs [hunk 2]
```diff
@@ -189,8 +194,8 @@ macro_rules! vertical {
 #[macro_export]
 macro_rules! horizontal {
     ($( $constraint:tt )+) => {
-        ratatui::prelude::Layout::default()
-            .direction(ratatui::prelude::Direction::Horizontal)
+        ratatui::layout::Layout::default()
+            .direction(ratatui::layout::Direction::Horizontal)
             .constraints($crate::constraints!( $($constraint)+ ))
```
Comment (EdJoPaTo) [inline src/layout.rs:199]:
```suggestion
        ratatui::layout::Layout::horizontal($crate::constraints!( $($constraint)+ ))
```

File: src/layout.rs [hunk 3]
```diff
@@ -166,8 +171,8 @@ macro_rules! constraints {
 #[macro_export]
 macro_rules! vertical {
     ($( $constraint:tt )+) => {
-        ratatui::prelude::Layout::default()
-            .direction(ratatui::prelude::Direction::Vertical)
+        ratatui::layout::Layout::default()
+            .direction(ratatui::layout::Direction::Vertical)
             .constraints($crate::constraints!( $($constraint)+ ))
```
Comment (EdJoPaTo) [inline src/layout.rs:176]:
```suggestion
        ratatui::layout::Layout::vertical($crate::constraints!( $($constraint)+ ))
```

**Comment by EdJoPaTo** [comment]
> I'm not a fan at all of this particular macro as it feels like "too much magic" to me, so I have no opinion on merging this.

In general, I also avoid using Layout in the first place. Its logic is too complicated and writing that math myself is short enough so I never wanted to care about that Layout magic.

There is simply too much going on for my taste. So I neither use layout nor would use this macro.

**Pull request merged**